src/main.rs

```diff
@@ -1,52 +1,94 @@
 extern crate docopt;
 extern crate rand;
 extern crate rustc_serialize;
-#[macro_use] extern crate lazy_static;
 
-use std::process;
+use std::{process, io, path};
+use std::fs::File;
+use std::io::Read;
 use rand::distributions::{IndependentSample, Range};
 use docopt::Docopt;
 
 const USAGE: &'static str = "
 Generate XKCD style passwords.
 
 Usage:
-  xkcdpass [-c <count>]
+  xkcdpass [options] [-w <list>...]
 
 Options:
-  -h --help     Show this screen.
-  -v --version  Show version.
-  -c <count>    Number of words [default: 4].
+  -h --help                 Show this screen.
+  -v --version              Show version.
+  -c <count>                Number of words [default: 4].
+  -w --use-wordlists <list>
 ";
 
-static WORDLIST: &'static str = include_str!("wordlist.txt");
-lazy_static! {
-    static ref MAX_OFFSET: usize = WORDLIST.trim().chars().filter(|c| *c as u32 == 0x0a).count();
-    static ref BETWEEN: Range<usize> = Range::new(0, *MAX_OFFSET);
-}
+static DEFAULT_WORDLIST: &'static str = include_str!("wordlist.txt");
 
 #[derive(Debug, RustcDecodable)]
 struct Args {
     flag_c: usize,
+    flag_w: Option<Vec<String>>,
+}
+
+fn load_wordlist(path: String) -> io::Result<String> {
+    let path = path::Path::new(&path);
+    let mut file = File::open(path)?;
+    let mut string = String::new();
+    file.read_to_string(&mut string)?;
+    Ok(string)
+}
+
+fn load_wordlists(paths: Vec<String>) -> io::Result<String> {
+    let mut size = 0;
+    let wordlists = paths.into_iter()
+        .map(|p| {
+            let list = load_wordlist(p);
+            if let Ok(ref s) = list {
+                size += s.len();
+            }
+            list
+        })
+        .collect::<Result<Vec<_>, _>>()?;
+
+    let mut string = String::with_capacity(size);
+
+    for list in wordlists {
+        string.push_str(&list);
+    }
```

Comment:
Instead of counting bytes and allocating a new string to hold all the old strings, you could also return a vector of strings. Later, when finding the max offset or the word, you can use iterator chaining:

See https://is.gd/DEPbAD for an example.